packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -242,6 +250,10 @@ class _DefaultPub implements Pub {
         'The time now is: $now'
       );
     }
+    // Insert references to synthetic flutter package.
+    if (yamlConfig.existsSync()) {
```

Comment:
I don't really like this, in theory adding a yaml wouldn't re-trigger pub get,  that would have to be done automatically